### PR TITLE
Update Go SDK to v0.10.15

### LINKED ISSUE #179: CLI hangs if domain doesn't exist
This hangs if domain sample doesn't exist: `./cadence -do sample workflow list`

### PR DESCRIPTION
**Pull request by alexshtin**
**What changed?**
Go SDK was updated to latest v0.10.15 (new way to create client and worker).

**Why?**
Time has come. It also fixes #179.

**How did you test it?**
Run integration tests.

**Potential risks**
Everything that uses Go SDK client might be broken.

### CODE REVIEW

File: cmd/server/temporal/server.go [hunk 1]
```diff
@@ -163,14 +162,17 @@ func (s *server) startService() common.Daemon {
 
 	if s.cfg.PublicClient.HostPort == "" {
 		log.Fatalf("need to provide an endpoint config for PublicClient")
-	} else if h, _, err := net.SplitHostPort(s.cfg.PublicClient.HostPort); err != nil || len(h) == 0 {
```
Comment (alexshtin) [inline cmd/server/temporal/server.go]: gRPC target can be `dns://mydomain.com:7233` which doesn't fit this validation.

File: cmd/server/temporal/server.go [hunk 2]
```diff
@@ -163,14 +162,17 @@ func (s *server) startService() common.Daemon {
...
-		log.Fatalf("Malformed PublicClient HostPort, must be host:port - '%v' - Error - %v", s.cfg.PublicClient.HostPort, err)
 	} else {
-		connection, err := rpc.Dial(s.cfg.PublicClient.HostPort)
+		var err error
+		params.PublicClient, err = sdkclient.NewClient(sdkclient.Options{
+			HostPort:     s.cfg.PublicClient.HostPort,
+			DomainName:   common.SystemLocalDomainName,
+			MetricsScope: params.MetricScope,
+			Tracer:       opentracing.GlobalTracer(), //???
```
Comment (alexshtin) [inline cmd/server/temporal/server.go]: @samarabbas, `opentracing.GlobalTracer()` was added in two places but all the rest did not have it. Is it harmful to always add it?

File: common/resource/resourceTest.go [hunk 3]
```diff
@@ -313,8 +311,8 @@ func (s *Test) GetWorkerServiceResolver() membership.ServiceResolver {
 // internal services clients
 
 // GetSDKClient for testing
-func (s *Test) GetSDKClient() workflowservice.WorkflowServiceClient {
-	return s.SDKClient
+func (s *Test) GetSDKClient() sdkclient.Client {
+	panic("user should implement this method for test")
```
Comment (alexshtin) [inline common/resource/resourceTest.go]: It should not be hard to generate mock for `sdkclient.Client` but it seems not to be used anywhere.

Comment (alexshtin) [reply common/resource/resourceTest.go]: I am laughing at myself 4 hours ago :-)

File: cmd/server/temporal/server.go (hunk 2 above)
Comment (samarabbas) [reply cmd/server/temporal/server.go]: I don't think it even matters as we don't do anything with it after moving to grpc on client side.

File: go.mod [hunk 4]
```diff
@@ -65,22 +65,22 @@ require (
...
 	go.uber.org/thriftrw v1.20.2
-	go.uber.org/zap v1.14.0
-	golang.org/x/net v0.0.0-20200202094626-16171245cfb2
+	go.uber.org/zap v1.14.1
+	golang.org/x/net v0.0.0-20200301022130-244492dfa37a
 	golang.org/x/oauth2 v0.0.0-20190226205417-e64efc72b421
-	golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
-	golang.org/x/tools v0.0.0-20191114200427-caa0b0f7d508
+	golang.org/x/time v0.0.0-20191024005414-555d28b269f0
+	golang.org/x/tools v0.0.0-20200317043434-63da46f3035e
```
Comment (samarabbas) [inline go.mod]: I think this dependency update is causing issues on buildkite.

**Pull request merged**